test/parseComment.js

```diff
@@ -294,11 +304,243 @@ describe('parseComment', function () {
+      problems: []
+    });
+  });
+
+  it('Handle inline type tag with "prefixed" text in description', function () {
+    const parsed = parseComment({
+      value: `* A link to [something awesome!]{@link Something}`
+    });
+    expect(parsed).to.deep.equal({
+      description: 'A link to [something awesome!]{@link Something}',
+      tags: [],
+      inlineTags: [
+        {
+          tag: 'link',
+          namepathOrURL: 'Something',
+          text: 'something awesome!',
+          textStyle: 'prefix',
+          start: 10,
+          end: 47
+        }
+      ],
+      source: [
+        {
+          number: 0,
+          source: '/** A link to [something awesome!]{@link Something}*/',
+          tokens: {
+            delimiter: '/**',
+            description: 'A link to [something awesome!]{@link Something}',
+            end: '*/',
+            lineEnd: '',
+            name: '',
+            postDelimiter: ' ',
+            postName: '',
+            postTag: '',
+            postType: '',
+            start: '',
+            tag: '',
+            type: ''
+          }
+        }
+      ],
+      problems: []
+    });
+  });
+
+  it('Handle plain inline type tag in tag', function () {
+    const parsed = parseComment({value: `* @see A link to {@link Something}`});
+    expect(parsed).to.deep.equal({
+      description: '',
+      tags: [
+        {
+          tag: 'see',
+          name: '',
+          type: '',
+          optional: false,
+          description: 'A link to {@link Something}',
+          problems: [],
+          inlineTags: [
+            {
+              tag: 'link',
+              namepathOrURL: 'Something',
+              text: '',
+              textStyle: 'plain',
+              start: 10,
+              end: 27
+            }
+          ],
+          source: [
+            {
+              number: 0,
+              source: '/** @see A link to {@link Something}*/',
+              tokens: {
+                delimiter: '/**',
+                description: 'A link to {@link Something}',
+                end: '*/',
+                lineEnd: '',
+                name: '',
+                postDelimiter: ' ',
+                postName: '',
+                postTag: ' ',
+                postType: '',
+                start: '',
+                tag: '@see',
+                type: ''
+              }
+            }
+          ]
+        }
+      ],
+      inlineTags: [],
+      source: [
+        {
+          number: 0,
+          source: '/** @see A link to {@link Something}*/',
+          tokens: {
+            delimiter: '/**',
+            description: 'A link to {@link Something}',
+            end: '*/',
+            lineEnd: '',
+            name: '',
+            postDelimiter: ' ',
+            postName: '',
+            postTag: ' ',
+            postType: '',
+            start: '',
+            tag: '@see',
+            type: ''
+          }
+        }
+      ],
+      problems: []
+    });
+  });
+
```

Comment:
To avoid further bloating the tests, I choose to implement just one of the text styles for the case of "inline tag in tag".